doc/how-to/getting_started_go.rst

```diff
@@ -1,256 +1,279 @@
 
-Update a :term:`field` value using ``Update``:
+..  literalinclude:: /code_snippets/snippets/connectors/go/hello.go
+    :language: go
+    :start-at: // Insert data
+    :end-before: // Select by primary key
+    :dedent:
 
-.. code-block:: go
+This code makes insert requests asynchronously:
 
-    resp, err = conn.Update("tester", "primary", []interface{}{4}, []interface{}{[]interface{}{"+", 2, 3}})
+-   The ``Future`` structure is used as a handle for asynchronous requests.
+-   The ``NewInsertRequest()`` method creates an insert request object that is executed by the connection.
 
-This increases by 3 the value of field ``2`` in the tuple with ``id = 4``.
-If a tuple with this ``id`` doesn't exist, Tarantool will return an error.
+..  NOTE::
 
-Now use ``Replace`` to totally replace the tuple that matches the
-primary key. If a tuple with this primary key doesn't exist, Tarantool will
-do nothing.
+    Making requests asynchronously is the recommended way to perform data operations.
+    Further requests in this tutorial are made synchronously.
 
-.. code-block:: go
 
-    resp, err = conn.Replace("tester", []interface{}{4, "New band", 2011})
+.. _getting_started_go_querying_data:
 
-You can also update the data using ``Upsert`` that works similarly
-to ``Update``, but creates a new tuple if the old one was not found.
+Querying data
+*************
 
-.. code-block:: go
+To get a tuple by the specified primary key value, use ``NewSelectRequest()`` to create an insert request object:
 
-    resp, err = conn.Upsert("tester", []interface{}{4, "Another band", 2000}, []interface{}{[]interface{}{"+", 2, 5}})
+..  literalinclude:: /code_snippets/snippets/connectors/go/hello.go
+    :language: go
+    :start-at: // Select by primary key
+    :end-at: Tuple selected the primary key value
+    :dedent:
 
-This increases by 5 the value of the third field in the tuple with ``id = 4``, or
-inserts the tuple ``(4, "Another band", 2000)`` if a tuple with this ``id``
-doesn't exist.
+You can also get a tuple by the value of the specified index by using ``Index()``:
 
-.. _getting_started-go-delete:
+..  literalinclude:: /code_snippets/snippets/connectors/go/hello.go
+    :language: go
+    :start-at: // Select by secondary key
+    :end-at: Tuple selected the secondary key value
+    :dedent:
 
-********************************************************************************
-Deleting data
-********************************************************************************
 
-To delete a tuple, use ``connection.Delete``:
 
-.. code-block:: go
+.. _getting_started_go_updating_data:
 
-    resp, err = conn.Delete("tester", "primary", []interface{}{4})
+Updating data
+*************
 
-To delete all tuples in a space (or to delete an entire space), use ``Call``.
-We'll focus on this function in more detail in the
-:ref:`next <getting_started-go-stored-procs>` section.
+``NewUpdateRequest()`` can be used to update a tuple identified by the primary key as follows:
```

Comment:
```suggestion
``NewUpdateRequest()`` updates a tuple identified by the primary key as follows:
```
Here and in some places below.
Less words, more direct expression. Anyway, up to you.

Reply:
Technically, NewUpdateRequest() doesn't update a tuple - it creates an object that is used by the connection to update a tuple. So, I'd keep it as is.